src/Nethermind/Nethermind.Core/ValidationResult.cs

```diff
@@ -0,0 +1,13 @@
+// SPDX-FileCopyrightText: 2024 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+namespace Nethermind.Core;
+
+public record struct ValidationResult(string? Error)
+{
+    public static ValidationResult Success => new(null);
+    public static implicit operator bool(ValidationResult result) => result.AsBool();
+    public static implicit operator ValidationResult(string error) => new(error);
+    public override string ToString() => Error ?? "Success";
+    public bool AsBool() => Error is null;
+}
```

Comment:
NOTE: We have several `*Result` types that could be replaced with a generic `Result<E, T>`

Reply:
Maybe but not sure, here we are using syntactic sugar with implicit conversion to bool, which allows us to treat it as a boolean value. Not sure if we can leverage that with generics easily and if it would be worth it - would have to see it.